### PR TITLE
Filsnap Demo

### PR DESCRIPTION
**Pull request by avivash**
# Description

- Add `/filfund` route that allows funds to be contributed to the widget via filsnap using native FIL

## Link to issue

Please add a link to any relevant issues/tickets

## Type of change

- [x] New feature (non-breaking change that adds functionality)

## Screencast
https://www.loom.com/share/aa6467fd17f8414590b0c71b660d631b?sid=d7ea6681-4808-4530-b186-0be555b732c3

## Staging Domain

https://fundring-filsnap.fission.app/

### CODE REVIEW

**Comment by hugomrdias** [comment]
it looks good but AFAICS its not actually using the filsnap API and the tx insights are not even returning anything (which is probably a bug).

File: src/routes/filfund/components/FundRingWidget/Connected.svelte [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+  const FUNDRING_CONTRACT_OWNER_ADDRESS =
+    't410ftgo7vbcywlxf72blsnr4tgmpvcxrs6icg34pouq'
+  const LOADING_TEXT = ['processing', 'sit tight', 'network speed may vary']
+  const RPC_ENDPOINT = 'https://api.calibration.node.glif.io/rpc/v1'
+
+  const rpc = new RPC({
+    api: RPC_ENDPOINT,
+    network: 'testnet'
+  })
```
Comment (hugomrdias) [inline src/routes/filfund/components/FundRingWidget/Connected.svelte:29]: The new adapter version has a `rpc()` method that returns a rpc client configured by the filsnap.

File: src/routes/filfund/components/FundRingWidget/Disconnected.svelte [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  export let snapConnected = null
+  export let walletConnected = null
+  export let title: string = 'Help fund us!'
+  export let description: string =
+    'If you rely upon Fund Ring’s efforts to keep your project going, please consider supporting our funding goal. Every little bit helps.'
+
+  let loading = false
+
+  const checkIfFlaskIsInstalled = async () => {
+    const hasFlask = await FilsnapAdapter.hasFlask()
```
Comment (hugomrdias) [inline src/routes/filfund/components/FundRingWidget/Disconnected.svelte:18]:
```suggestion
    const hasFlask = await FilsnapAdapter.hasSnaps()
```

File: src/routes/filfund/components/FundRingWidget/Disconnected.svelte [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    }
+  }
+
+  // Prompt the user to connect their FilSnap wallet client
+  const handleConnect = async () => {
+    loading = true
+    try {
+      await checkIfFlaskIsInstalled()
+
+      const isSnapConnected = await FilsnapAdapter.isConnected()
```
Comment (hugomrdias) [inline src/routes/filfund/components/FundRingWidget/Disconnected.svelte:31]:
```suggestion
      const isSnapConnected = await FilsnapAdapter.isAvailable()
```

**Review by hugomrdias — CHANGES_REQUESTED** [review]
Please update adapter to v1.0.0 and use "Filecoin Wallet" in user facing messages.

File: src/routes/filfund/components/FundRingWidget/Connected.svelte [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+  }
+
+  onMount(async () => {
+    fetchingData = true
+    loading = true
+
+    snap = await FilsnapAdapter.connect(
+      { network: 'testnet' },
+      'npm:filsnap',
+      '*'
```
[the remark this answers is not in the document]
Comment (avivash) [reply src/routes/filfund/components/FundRingWidget/Connected.svelte]: updated 👍🏼

**Comment by avivash** [comment]
thanks @hugomrdias! I've updated the PR as per your feedback 👍🏼

**Review by hugomrdias — APPROVED** [review]
LGTM

**Pull request merged**